apps/web/composables/useBlockManager/useBlockManager.ts

```diff
@@ -108,7 +108,13 @@ export const useBlockManager = () => {
     isEditingEnabled.value = !deepEqual(cleanData.value, data.value);
   };
 
-  const isLastBlock = (index: number) => index === data.value.length - 1;
+  const isLastBlock = (index: number) => {
+    if (!data.value || data.value.length === 0) return false;
+    const lastNonFooterIndex = [...data.value].reverse().findIndex((block) => block.name !== 'Footer');
+    if (lastNonFooterIndex === -1) return false;
+    const actualLastNonFooterIndex = data.value.length - 1 - lastNonFooterIndex;
+    return index === actualLastNonFooterIndex;
```

Comment:
This reads complicated.

- Why are we reversing the blocks list?
- Why do we have to subtract from `length` twice?

Isn't it enough to check `index === data.value.length - 2` if there's a footer, and `- 1` if there's no footer?